test/wadler_test.rb

```diff
@@ -0,0 +1,155 @@
+# frozen_string_literal: true
+
+require 'prettyprint'
+
+require_relative 'lib'
+
+def test_lib_with_ref(margin, expected, builder_block)
+  _(builder_block.call(margin)).must_equal expected
+  _(builder_block.call(margin, false)).must_equal expected
+end
```

Comment:
1. It might be better, instead of using boolean flags to know which lib we're testing, to simply send the instance: Oppen then PrettyPrint.
1. Don't forget to add test message next to each line. Makes understanding which part diverged, from the test run screen, easier to read.
1. `test_lib_with_ref` -> `check_roundtrip`.